### PR TITLE
Move header type to neqo-qpack, reexport it through neqo-http3.

### LINKED ISSUE #137: Consider having a Headers type ?
As explained in https://github.com/mozilla/neqo/pull/131#issuecomment-520134820 It might be useful to:

Move the Header type to the common crate.
Have only one Headers type, which would be `Vec<Header>`.

Replace uses of `&[Header]` (or `&[(String,String)]`) to use the new Headers type.

Not sure using a Vec instead of a slice ref for fix sized arrays has a performance impact.

What's your opinion ? :)

### PR DESCRIPTION
**Pull request by o0Ignition0o**
fix #137

### CODE REVIEW

**Comment by agrover** [comment]
This suggests a `Headers` type is a `Vec<Header>` would also be handy.

File: neqo-client/src/main.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #![deny(warnings)]
 use neqo_common::Datagram;
 use neqo_crypto::init_db;
+use neqo_http3::request_stream_server::Header;
```
Comment (agrover) [inline neqo-client/src/main.rs]: re-export `Header` at top of http3 so apps can just do `use neqo_http3::Header`?

**Comment by o0Ignition0o** [comment]
Oh I think I just enraged Chronos, the god of monotonic time ^^

**Comment by o0Ignition0o** [comment]
Not really sure what's going on here :(

File: neqo-crypto/src/time.rs [hunk 2]
```diff
@@ -180,7 +180,7 @@ mod test {
 
     #[test]
     fn convert_stable() {
-        let now = Time::from(Instant::now());
+        let now = Time::now();
```
Comment (agrover) [inline neqo-crypto/src/time.rs]: this looks like an unrelated change that may be the source of the CI trouble, so just omit from this PR please.

**Comment by o0Ignition0o** [comment]
I have actually just added this change to retrigger the CI.

I think there's a weird bug with timestamps which has been unfortunately
caught during my try.

I'll revert the commit, can you rerun the CI then ? ^^

**Comment by o0Ignition0o** [comment]
There you go @agrover 
Btw should I open an issue referencing this build fail ? It would be nice to figure it out, given the number of Firefox users I'm pretty sure it will happen again someday.

**Comment by agrover** [comment]
@o0Ignition0o yes please open an issue. FWIW I cannot reproduce locally.

As far as I'm concerned this PR can be merged once conflicts are resolved.

**Comment by o0Ignition0o** [comment]
MB, forgot to update a test :(

**Pull request merged**